### PR TITLE
Add Solar chat model

### PR DESCRIPTION
**Pull request by Tokkiu**
# Description
PR to add the `Solar` class

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [x] Added new notebook (that tests end-to-end)
- [ ] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-integrations/llms/llama-index-llms-solar/.gitignore
Comment (jaeholee2212) [inline llama-index-integrations/llms/llama-index-llms-solar/.gitignore:1]: why do we need this file?

Comment (nerdai) [reply llama-index-integrations/llms/llama-index-llms-solar/.gitignore:1]: these files are all included when using our cli tool to create a new integration package. .gitignore is mostly boiler plate, `pyproject.toml` is required to build the project as well as for dep management. `Makefile` is used for lilnting and formatting and testing convience

**Comment by nerdai** [comment]
@Tokkiu please run `make lint` and push the changes when you get the chance.

**Comment by Tokkiu** [comment]
> @Tokkiu please run `make lint` and push the changes when you get the chance.

thanks, will fix this asap.

**Comment by Tokkiu** [comment]
@nerdai Pls give a review if you have time. ❤️

**Review by logan-markewich — APPROVED** [review]

**Pull request merged**